sample-app/src/components/SearchBar.tsx

```diff
@@ -76,50 +114,70 @@ export default function SearchBar({
+    if (!matchedSubstrings || matchedSubstrings.length === 0) {
+      return <span>{value}</span>;
+    }
+    return renderHighlightedValue({ value, matchedSubstrings });
+  }
+
   return (
-    <div className='SearchBar'>
-      <InputDropdown
-        inputValue={query}
-        placeholder={placeholder}
-        screenReaderInstructions={SCREENREADER_INSTRUCTIONS}
-        screenReaderInstructionsId={screenReaderInstructionsId}
-        screenReaderText={screenReaderText}
-        onSubmit={executeQuery}
-        onInputChange={value => {
-          answersActions.setQuery(value);
-        }}
-        onInputFocus={() => {
-          autocompletePromiseRef.current = executeAutocomplete();
-        }}
-        renderButtons={renderSearchButton}
-        cssClasses={{
-          dropdownContainer: 'Autocomplete',
-          inputElement: 'SearchBar__input',
-          inputContainer: 'SearchBar__inputContainer'
-        }}
-      >
-        {
-          options.length > 0 &&
-          <DropdownSection
-            options={options}
-            optionIdPrefix='Autocomplete__option-0'
-            onFocusChange={value => {
-              answersActions.setQuery(value);
-            }}
-            onClickOption={option => {
-              answersActions.setQuery(option.value);
-              executeQuery();
-            }}
-            cssClasses={{
-              sectionContainer: 'Autocomplete__dropdownSection',
-              sectionLabel: 'Autocomplete__sectionLabel',
-              optionsContainer: 'Autocomplete_sectionOptions',
-              option: 'Autocomplete__option',
-              focusedOption: 'Autocomplete__option--focused'
-            }}
-          />
-        }
-      </InputDropdown>
+    <div className={cssClasses.container}>
+      <div className={cssClasses.inputDropdownContainer}>
+        <InputDropdown
+          inputValue={query}
+          placeholder={placeholder}
+          screenReaderInstructions={SCREENREADER_INSTRUCTIONS}
+          screenReaderInstructionsId={screenReaderInstructionsId}
+          screenReaderText={screenReaderText}
+          onSubmit={executeQuery}
+          onInputChange={value => {
+            answersActions.setQuery(value);
+          }}
+          onInputFocus={() => {
+            autocompletePromiseRef.current = executeAutocomplete();
+          }}
+          renderLogo={() => <YextLogoIcon />}
+          renderSearchButton={renderSearchButton}
+          cssClasses={cssClasses}
+        >
+          {
+            options.length > 0 &&
+            <DropdownSection
+              options={options}
+              optionIdPrefix='Autocomplete__option-0'
+              onFocusChange={value => {
+                answersActions.setQuery(value);
+              }}
+              onClickOption={option => {
+                answersActions.setQuery(option.value);
+                executeQuery();
+              }}
+              cssClasses={cssClasses}
```

Comment:
Yes, that's a downside of this approach that I discussed with Tom when we decided to go this direction. In some situations, we can pass the sub components to the parent components so that the css classes of the sub component is no longer part of the class interface of the parent. The `InputDropdown` is controlled by the `SearchBar` so that approach wouldn't work so great here, but it might work well in other situations. 

I think the manual bookkeeping makes the most sense for now. If there are conflicts, we can namespace the conflicting names for the subcomponents. These class interfaces shouldn't be too large for most components, so I think it will be pretty manageable. If we run into problems with that approach, there are more nuclear solutions such as name-spacing every classname with the module name (e.g. container becomes `SearchBar_container`) or separating the classes object by subcomponent as you suggested